python_modules/dagster/dagster/_core/storage/legacy_storage.py

```diff
@@ -488,6 +488,13 @@ def get_materialization_count_by_partition(
             asset_keys, after_cursor
         )
 
+    def get_latest_asset_materialization_planned_event_by_partition(
```

Comment:
again would prefer to not have to add new storage method if possible, by refactoring the existing `get_materialization_count_by_partition`